autogpt/llm/llm_utils.py

```diff
@@ -293,3 +293,24 @@ def create_embedding(
     )  # normalize the length to one
     chunk_embeddings = chunk_embeddings.tolist()
     return chunk_embeddings
+
+
+def check_smart_llm_model_access() -> None:
+    """Check if smart model is available for use. If not, set to fast model."""
+    cfg = Config()
+
+    messages = [
+        {"role": "user", "content": ""},
+    ]
+    try:
+        create_chat_completion(
+            model=cfg.smart_llm_model, messages=messages, temperature=0
+        )
+    except InvalidRequestError:
+        logger.typewriter_log(
+            "WARNING: ",
+            Fore.YELLOW,
+            f"You do not have access to {cfg.smart_llm_model}. Setting default "
+            f"to {cfg.fast_llm_model}.",
+        )
+        cfg.smart_llm_model = cfg.fast_llm_model
```

Comment:
What if they have set for both fast and smart GPT4?

Reply:
I've moved the function into the configurator - if both flags are added and chat gpt check fails, it updates gpt3only to True and gpt4only to False. There's also the same log warning.

Reply:
Hey @k-boikov, so I've updated the approach to checking both models individually, and updating to "gpt-3.5-turbo" if either them raises an exception. Is this what you had in mind?

Reply:
Hey @k-boikov, changes made:

- check_models() fn added which takes model name and model type, then returns original model name if valid, or hard-coded "gpt-3.5-turbo" if not valid
- Models are retrieved from get_models() from APIManager, which uses openai.Model.list()
- Checks are done at top of create_config()

Let me know if you have any other changes.